File: gitdot-api-ts/src/endpoint/repository/get-permission.ts
import { z } from "zod";
import { RepositoryPermissionResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetPermissionRequest = z.object({});
export type GetPermissionRequest = z.infer<typeof GetPermissionRequest>;

export const GetPermissionResponse = RepositoryPermissionResource;
export type GetPermissionResponse = z.infer<typeof GetPermissionResponse>;

export const GetPermission = {
  path: "/repository/{owner}/{repo}/permission",
  method: "GET",
  request: GetPermissionRequest,
  response: GetPermissionResponse,
} as const satisfies Endpoint;
export type GetPermission = typeof GetPermission;

File: gitdot-api/src/endpoint/repository/get_permission.rs
use crate::{endpoint::Endpoint, resource::repository::RepositoryPermissionResource};

pub struct GetPermission;

impl Endpoint for GetPermission {
    const PATH: &'static str = "/repository/{owner}/{repo}/permission";
    const METHOD: http::Method = http::Method::GET;

    type Request = ();
    type Response = GetPermissionResponse;
}

pub type GetPermissionResponse = RepositoryPermissionResource;
